components/merkledb/src/db.rs

```diff
@@ -614,6 +705,19 @@ impl Fork {
     pub fn working_patch(&self) -> &WorkingPatch {
         &self.working_patch
     }
+
+    /// Returns snapshot that also captures flushed changes in the fork,
+    /// but does not capture unflushed changes.
+    pub fn snapshot_without_unflushed_changes(&self) -> &dyn Snapshot {
```

Comment:
What about `snapshot_with_flushed_changes` ? Weird name as well but a little bit shorter 🙃

Reply:
It was initially named that way; see https://github.com/exonum/exonum/pull/1523#discussion_r342094796.